### PR TITLE
feat(components/indicators): add ability to set aria controls and expanded states on the help inline component

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#2340587](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2340587)
[AB#2255356](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2255356)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1225/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/a11y-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/ag-grid-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/core-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/lists-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1225/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1225/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1225/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1225/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1225/playground/)

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.spec.ts [hunk 1]
```diff
@@ -30,12 +30,95 @@ describe('Help inline component', () => {
...
     fixture.detectChanges();
-    expect(cmp.buttonIsClicked).toBe(true);
+    expect(cmp.showHelpText).toBe(true);
   });
 
-  it('should pass accessibility', async () => {
+  it('should pass accessibility (ariaControls: undefined, ariaExpanded: undefined)', async () => {
     fixture.detectChanges();
     await fixture.whenStable();
+    expect(
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/indicators/src/lib/modules/help-inline/help-inline.component.spec.ts]: These tests all have similar contents, just with what's expected changing between each. Factor out into a function that accepts the expected values (use `null` with `toBe(null)` when parameterizing the new function). Also factor out the call to `fixture.nativeElement.querySelector('.sky-help-inline')` so you're not calling it repeatedly.

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts [hunk 2]
```diff
@@ -1,17 +1,52 @@
...
+    this.#_ariaControls = value;
+    this.ariaExpandedComputed = value ? !!this.ariaExpanded : null;
+  }
+
+  public get ariaControls(): string | undefined {
+    return this.#_ariaControls;
+  }
+
+  /**
+   * Whether an element exposed by the help inline button is exposed. When set to `true`, the property will only be applied to the HTML element when the `ariaControls` input is also set.
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: I'd change this to "Whether an element controlled by the help inline button is expanded".

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 
 @Component({
   selector: 'sky-help-inline',
   templateUrl: './help-inline.component.html',
   styleUrls: ['./help-inline.component.scss'],
 })
 export class SkyHelpInlineComponent {
+  /**
+   * The ID to identify the element that the help inline button exposes.
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: I'd change this to "The ID of the element that the help inline button controls."

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts [hunk 4]
```diff
@@ -1,17 +1,52 @@
...
+   * This property [supports accessibility rules for disclosures](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure).
+   * For more information about the `aria-controls` attribute, see the [WAI-ARIA definition](https://www.w3.org/TR/wai-aria/#aria-controls).
+   */
+  @Input()
+  public set ariaControls(value: string | undefined) {
+    this.#_ariaControls = value;
+    this.ariaExpandedComputed = value ? !!this.ariaExpanded : null;
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: I think a pipe for `aria-expanded` be a cleaner implementation. That way you could use simple public properties without `get`/`set` accessors and put all the logic in the pipe.

```
function transform(ariaExpanded: boolean | undefined, ariaControls: string | undefined): boolean | null {
  return ariaControls ? !!ariaExpanded : null;
}
```

```
[attr.aria-expanded]="ariaExpanded | skyHelpInlineAriaExpanded:ariaControls"
```

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: Done

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts (hunk 4 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: Done

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/indicators/src/lib/modules/help-inline/help-inline.component.ts]: Done

File: libs/components/indicators/src/lib/modules/help-inline/help-inline.component.spec.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/indicators/src/lib/modules/help-inline/help-inline.component.spec.ts]: Done

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See null

**Pull request merged**